docs/data/material/components/autocomplete/UseAutocomplete.tsx

```diff
@@ -20,11 +19,11 @@ const Listbox = styled('ul')(({ theme }) => ({
   zIndex: 1,
   position: 'absolute',
   listStyle: 'none',
-  backgroundColor: theme.palette.background.paper,
+  backgroundColor: theme.palette.mode === 'light' ? '#fff' : '#000',
   overflow: 'auto',
   maxHeight: 200,
   border: '1px solid rgba(0,0,0,.25)',
-  [`& li.${autocompleteClasses.focused}`]: {
+  '& li.Mui-focused': {
```

Comment:
It's hardcoded anyway https://github.com/mui/material-ui/blob/64e5dc1656e2c70e1b692618a5121ff67c645e78/packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js#L353

Reply:
useAutocomplete needs an overhaul to match the other hooks in Base. It uses different patterns now (like setting a class name, that doesn't happen anywhere else in other hooks).

Reply:
@michaldudak  What would be the alternative in this case? For context: #26181

> needs an overhaul to match the other hooks in Base.

I think that it would be great to review https://github.com/mui/mui-x/pull/5504. I wanted to raise how the unstyled hook API feels a bit different from our useAutocomplete, which is based on https://www.downshift-js.com/use-combobox.

Reply:
@michaldudak Bump. I'm curious about two things:

1. How is the pattern hook different now, e.g. developers seem to enjoy the current approach: https://twitter.com/shyam_lohar_/status/1571000213388791808 (which also makes the case for a headless first approach, anytime we create a new component, in Joy UI, Material UI, MUI X, we should likely start from the headless DX).
2. How to align with MUI X on it, e.g. with @flaviendelangle #5504